web/src/pages/Users.tsx

```diff
@@ -1,59 +1,131 @@
+                .then((updatedUser) => {
+                    if (updatedUser) {
+                        requestUsers();
+                    }
+                })
+                .catch((error) => {
+                    console.error('Erro ao desativar o usuário:', error);
+                });
+        }
+    };
+
+    const handleActivateUser = (user: UserSchema) => {
+        const confirmResult = window.confirm('Tem certeza de que deseja ativar este usuário?');
+        if (confirmResult) {
+            const updatedUserData = {
+                id: user.id,
+                name: user.name,
+                registration: user.registration,
+                userType: user.userType,
+                email: user.email,
+                password: user.password,
+                insertDate: user.insertDate,
+                expiredDate: user.expiredDate
+            };
+            putUser(user.id, updatedUserData, 2)
+                .then((updatedUser) => {
+                    if (updatedUser) {
+                        requestUsers();
+                    }
+                })
+                .catch((error) => {
+                    console.error('Erro ao ativar o usuário:', error);
+                });
+        }
+    };
+
     const columns: ColumnsType<UserSchema> = [
         {
-          title: 'Nome',
-          dataIndex: 'name',
-          key: 'name',
+            title: 'Nome',
+            dataIndex: 'name',
+            key: 'name',
+        },
+        {
+            title: 'Matrícula',
+            dataIndex: 'registration',
+            key: 'registration',
         },
         {
-          title: 'Matrícula',
-          dataIndex: 'registration',
-          key: 'registration',
+            title: 'Email',
+            dataIndex: 'email',
+            key: 'email',
         },
         {
-          title: 'Email',
-          dataIndex: 'email',
-          key: 'email',
+            title: 'Função',
+            dataIndex: 'userType',
+            key: 'userType',
         },
         {
-          title: 'Função',
-          dataIndex: 'userType',
-          key: 'userType',
+            title: 'Status',
+            dataIndex: 'active',
+            key: 'active',
         },
         {
-          dataIndex: 'tags',
-          key: 'tags',
-          render: (_,data) => (data? <button>Ver</button> : null)
+            title: 'Ação',
+            dataIndex: 'expiredDate',
+            key: 'expiredDate',
+            render: (expiredDate, data) => (
+                expiredDate !== "N/A" ? (
+                    <button onClick={() => handleActivateUser(data)}>Ativar</button>
+                ) : (
+                    <button onClick={() => handleDeactivateUser(data)}>Desativar</button>
+                )
+            ),
         },
     ];
-    return (
+
+    const filteredUsers = showDeactivated
+    ? users.filter((user) => user.expiredDate !== "N/A")
+    : users.filter((user) => user.expiredDate === "N/A");
```

Comment:
podemos reduzir a quantidade de operações se reescrevermos o filtro de seguinte forma:

```suggestion
    const filteredUsers = users.filter((user) => showDeactivated == (user.expiredDate !== "N/A"));
```

| showDeactivated | user.expiredDate !== "N/A" | resultado |
|---------------------------|----------------------------------------|---------------|
| true | true | true | 
| false | false | true |
| true | false | false |
| false | true | false |

- mostrar desativados && possui expireDate = inclui user
- ocultar desativados && não possui expireDate = inclui user
- mostrar desativados && não possui expireDate = não inclui user
- ocultar desativados && possui expireDate = não inclui user